security/fma-holocene.md

```diff
@@ -0,0 +1,117 @@
+- (Execution Layer) L2ToL1MessagePasser Storage Root in Header
+- (Smart Contracts) Update to the MIPS contract
+
+Each change has it's own section below with a list of Failure Modes.
+
+Below are references for this project:
+
+- [PID: Holocene hardfork upgrade](https://www.notion.so/PID-Holocene-hardfork-upgrade-00ee1ffc414a407088fdb49841771527?pvs=21)
+- [Github tracker](https://github.com/orgs/ethereum-optimism/projects/84/views/6)
+- [Specs](https://specs.optimism.io/protocol/holocene/derivation.html?highlight=holocene#holocene-derivation)
+
+
+##  Holocene Derivation
+
+### The batcher violates the new (stricter) ordering rules
+
+#### Description
+
+If the batcher:
+  - sends transactions out of order, or
+   - orders frames within a transaction out of order
+
+This will lead to the consensus client hitting error conditions when it loads the frames from L1. According to the spec, frames will be dropped in order to maintain a contiguous frame queue. If those frames are not re-sent in a timely manner on L1, the safe chain would halt.
+    
+#### Risk Assessment
+ medium severity / medium likelihood
+#### Mitigations
+The batcher implementation could:
+  - leverage nonce management to avoid sending transactions out of order in the first place (see [spec](https://specs.optimism.io/protocol/holocene/derivation.html?highlight=holocene#batcher-hardening))
+  - be hardened so detect the chain halt and resubmit the frames which were dropped (see [spec](https://specs.optimism.io/protocol/holocene/derivation.html?highlight=holocene#batcher-hardening)) either:
+    - as a part of normal operation
+    - as a part of its startup behavior (i.e. to be triggered by a restart)
+    - as a part of an emergency mode triggered by an admin API, this could allow for manual intervention
+- continually check for contiguity in its internal state, and panic if this is violated (possibly then triggering some recovery behavior)
+#### Detection
+L2 safe chain halt
+#### Recovery Path(s)
+We would need to get the batcher to resubmit the transaction sent out of order. No governance or hardfork needed to recover.
+
+### Derivation Deadlock (either specification or op-node bug)
+
+#### Description
+It is possible that either i) the Holocene Derivation Specification or ii) the consensus client implementation has overlooked a corner case such that derivation simply halts when that corner case arises.
+#### Risk Assessment
+** high severity / medium likelihood
+#### Mitigations 
+Multi client testing should help surface and consensus client bugs, possibly including fuzzing of some description.
+We should backport as much insight as possible from the implementation (which is code) to the specs (which is just words). Here's an example of doing just that [holocene: fix frame queue specs of derivation](https://github.com/ethereum-optimism/specs/pull/386).
+#### Detection
+L2 safe chain halt
+#### Recovery Path(s):
+We would need to fix the the implementation via a hardfork.
```

Comment:
As a hotfix, we would probably also make an emergency release which migrates back to the old DP at the block where derivation is stuck. We might already implement something like this as a contingency, e.g. adding a `holocene_deactivation_time` that would then move back to old derivation if this time is set. We could then instruct node operators to set this flag to some value, providing a quick recovery path.